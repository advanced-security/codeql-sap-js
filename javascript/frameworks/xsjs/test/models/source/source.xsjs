/* ========== 1. Web Request ========== */

var webRequest1 = $.request;
var webRequest2 = $.request.entities[0];

/* ========== 1-1. Web Request Bodies ========== */

var webRequestBody1 = webRequest1.body;
var webRequestBody2 = webRequest2.body;

/* ========== 1-2. Web Request Parameters ========== */

var webRequestParam1 = webRequest1.parameters;
var webRequestParam2 = webRequest2.parameters;

/* ========== 1-3. Web Request Headers ========== */

var webRequestHeader1 = webRequest1.headers;
var webRequestHeader2 = webRequest2.headers;

/* ========== 2. Web Response ========== */

var webResponse1 = $.response;
var webResponse2 = $.response.entities[0];

/* ========== 2-1. Web Response Bodies ========== */

var webResponseBody1 = webResponse1.body;
var webResponseBody2 = webResponse2.body;

/* ========== 3. Mail, SMTPConnection ========== */

var mail = new $.net.Mail();
var smtpConnection = new $.net.SMTPConnection();

/* ========== 4. HTTP Client ========== */

var httpClient = new $.net.http.Client();

/* ========== 1. Retrieving Web Request Body ========== */

var webRequestBody11 = webRequestBody1.asArrayBuffer();
var webRequestBody12 = webRequestBody1.asString();
var webRequestBody13 = webRequestBody1.asWebRequest();

var webRequestBody21 = webRequestBody2.asArrayBuffer();
var webRequestBody22 = webRequestBody2.asString();
var webRequestBody23 = webRequestBody2.asWebRequest();

/* ========== 2. Retrieving Web Request Parameter Value ========== */

var webRequestParam11 = webRequestParam1.get("key");
var webRequestParam12 = webRequestParam1.key;

var webRequestParam11 = webRequestParam2.get("key");
var webRequestParam12 = webRequestParam2.key;

/* ========== 3. Receiving Response through HTTPClient ========== */

var inboundResponse = httpClient.getReponse();

var inboundResponseBody1 = inboundResponse.body;
var inboundResponseBody2 = inboundResponse.body.asArrayBuffer();
var inboundResponseBody3 = inboundResponse.body.asString();
var inboundResponseBody4 = inboundResponse.body.asWebRequest();

var inboundResponseData1 = inboundResponse.cacheControl;
var inboundResponseData2 = inboundResponse.contentType;
var inboundResponseData3 = inboundResponse.cookies;
var inboundResponseData4 = inboundResponse.entities;
var inboundResponseData5 = inboundResponse.headers;
var inboundResponseData6 = inboundResponse.status;
