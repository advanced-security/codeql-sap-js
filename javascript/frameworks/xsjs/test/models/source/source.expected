| source.xsjs:42:24:42:54 | webRequ ... uffer() | Remote flow source of type: Remote flow |
| source.xsjs:42:24:42:54 | webRequ ... uffer() | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:43:24:43:49 | webRequ ... tring() | Remote flow source of type: Remote flow |
| source.xsjs:43:24:43:49 | webRequ ... tring() | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:44:24:44:53 | webRequ ... quest() | Remote flow source of type: Remote flow |
| source.xsjs:44:24:44:53 | webRequ ... quest() | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:46:24:46:54 | webRequ ... uffer() | Remote flow source of type: Remote flow |
| source.xsjs:46:24:46:54 | webRequ ... uffer() | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:47:24:47:49 | webRequ ... tring() | Remote flow source of type: Remote flow |
| source.xsjs:47:24:47:49 | webRequ ... tring() | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:48:24:48:53 | webRequ ... quest() | Remote flow source of type: Remote flow |
| source.xsjs:48:24:48:53 | webRequ ... quest() | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:52:25:52:44 | webRequestParam1.get | Remote flow source of type: Remote flow |
| source.xsjs:52:25:52:44 | webRequestParam1.get | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:52:25:52:51 | webRequ ... ("key") | Remote flow source of type: Remote flow |
| source.xsjs:52:25:52:51 | webRequ ... ("key") | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:53:25:53:44 | webRequestParam1.key | Remote flow source of type: Remote flow |
| source.xsjs:53:25:53:44 | webRequestParam1.key | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:55:25:55:44 | webRequestParam2.get | Remote flow source of type: Remote flow |
| source.xsjs:55:25:55:44 | webRequestParam2.get | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:55:25:55:51 | webRequ ... ("key") | Remote flow source of type: Remote flow |
| source.xsjs:55:25:55:51 | webRequ ... ("key") | Remote flow source of type: Source node (remote) [from data-extension] |
| source.xsjs:56:25:56:44 | webRequestParam2.key | Remote flow source of type: Remote flow |
| source.xsjs:56:25:56:44 | webRequestParam2.key | Remote flow source of type: Source node (remote) [from data-extension] |
